base-notebook/Dockerfile

```diff
@@ -88,8 +88,12 @@ RUN conda install --quiet --yes \
     jupyter labextension install @jupyterlab/hub-extension@^0.8.1 && \
     npm cache clean --force && \
     rm -rf $CONDA_DIR/share/jupyter/lab/staging && \
-    rm -rf /home/$NB_USER/.cache/yarn && \
-    fix-permissions $CONDA_DIR && \
+    rm -rf /home/$NB_USER/.cache/yarn
+
+# Initialize configuration file
+RUN jupyter notebook --generate-config
+
+RUN fix-permissions $CONDA_DIR && \
```

Comment:
Thanks for making the change. Theres's one new issue now: The `fix-permissions` needs to be done in the same `RUN` command. If it's done separately, all of the files that have their permissions changed (i.e., everything in the home and conda directories) are duplicated in full across the docker image layers, doubling the size of the original layer in the final image.

Something like the following should do the trick:

```
# Install Jupyter Notebook, Lab, and Hub
# Generate a notebook server config
# Cleanup temporary files
# Correct permissions
# Do all this in a single RUN command to avoid duplicating all of the
# files across image layers when the permissions change
RUN conda install --quiet --yes \
    'notebook=5.6.*' \
    'jupyterhub=0.8.*' \
    'jupyterlab=0.32.*' && \
    conda clean -tipsy && \
    jupyter labextension install @jupyterlab/hub-extension@^0.8.1 && \
    npm cache clean --force && \
    jupyter notebook --generate-config && \
    rm -rf $CONDA_DIR/share/jupyter/lab/staging && \
    rm -rf /home/$NB_USER/.cache/yarn && \
    fix-permissions $CONDA_DIR && \
    fix-permissions /home/$NB_USER
```

(Note: Notebook 5.6 is out with a fix for a critical vulnerability in its bundled mathjax, so it'd be great to bump that version # while we're in here too.)

If you want to make this change, that's great. If not, I can add a commit that does it to your pull request if you check the "Allow edits from maintainers" checkbox in the GitHub pull request screen.